google-cloud/install-cli/mint-leaf.yml

```diff
@@ -0,0 +1,65 @@
+name: google-cloud/install-cli
+version: 1.0.0
+description: Install the Google Cloud SDK CLI
+
+parameters:
+  cli-version:
+    description: "Version of the CLI to install"
+    default: "latest"
+  components:
+    description: "Optional space-separated list of CLI components to install"
+    default: ""
+
+tasks:
+  - key: install-cli
+    run: |
+      architecture=""
+      case $(uname -m) in
+        x86_64) architecture="x86_64" ;;
+        arm|aarch64) architecture="arm" ;;
+        *) echo "Unsupported architecture: $(uname -m)"; exit 1 ;;
+      esac
+
+      cli_version="${{ params.cli-version }}"
+      if [[ "$cli_version" == "latest" ]]; then
+        echo
+        echo "Resolving latest version of Google Cloud SDK"
+        curl -o latest-cloud-sdk.tar.gz https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz
+        cli_version=$(tar -Ozxf latest-cloud-sdk.tar.gz google-cloud-sdk/VERSION)
+        rm latest-cloud-sdk.tar.gz
+        echo "Using version ${cli_version}"
+      fi
```

Comment:
Google really seem to want you to programatically find the latest version of the gcloud CLI. Alternatives include parsing the release notes HTML and requiring a specific version parameter.

The file downloaded here is only a portion of the gcloud CLI, currently about 39M vs. 140M for the architecture-/OS-targeted versioned file.